contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
-      model:
+        This includes for example:
+        - The beam device's or instrument's model, company, serial number, construction year, etc.
+        - Used software or code
+        - Experiment descriptive parameters as reference frames, resolution, calibration
+        - Photon beams with their respective properties such as angles and polarization
+        - Various optical beam path devices, which interact, manipulate or measure optical beams
+        - Characteristics of the medium surrounding the sample
+        - "Beam devices" for a beam path description
+        - Stages(NXmanipulator)
+        - Sensors and actuators to control or measure sample or beam properties
+      device_information(NXfabrication):
         doc: |
-          The name of the instrument.
-        \@version:
+          General device information of the optical spectroscopy setup, if
+          suitable (e.g. for a tabletop spectrometer or other non-custom build setups).
+          For custom build setups, this may be limited to the construction year.
+        exists: recommended
+        vendor:
+          exists: recommended
+        model:
+          exists: recommended
+        identifier:
+          exists: recommended
+        construction_year(NX_DATE_TIME):
+          exists: recommended
           doc: |
-            The used version of the hardware if available. If not a commercial
-            instrument use date of completion of the hardware.
-      company:
+            Datetime of setup's initial construction. This refers to the date of
+            first measurement after new construction  or relocation of the setup.
+            Should be a ISO8601 date/time stamp. It is recommended to add an explicit time zone,
+            otherwise the local time zone is assumed per ISO8601.
+      software_TYPE(NXprogram):
+        exists: recommended
+        program:
+          doc: |
+            Commercial or otherwise defined given name of the program that was
+            used to control any parts of the optical spectroscopy setup.
+            The uppercase TYPE should be replaced by a specification name, i.e.
+            "software_detector" or "software_stage" to specify the respective
+            program or software components.
+          \@version:
+            doc: |
+              Either version with build number, commit hash, or description of a
+              (online) repository where the source code of the program and build
+              instructions can be found so that the program can be configured in
+              such a way that result files can be created ideally in a
+              deterministic manner.
+          \@url:
+            exists: optional
+            doc: |
+              Description of the software by persistent resource, where the program,
+              code, script etc. can be found.
+      # general properties and reference frames
+      reference_frames(NXcoordinate_system_set):
```

Comment:
I suggest to move this out of the instrument (i.e., under top-level NXentry).

Reply:
yea, better suited there.